MAX! binding improvements

marcelrv opened the pull request:
* Discovery of multiple Cubes on the network
* Integration of MaxCubeDiscover.java into MaxCubeBridgeDiscovery.java for faster discovery results
* Improvement of discovery via mulicast message https://github.com/openhab/openhab/pull/1902
* Implementation of refreshType command to trigger refresh
* Change the numeric values types to BigDecimal to enable (once issue #64 fixed) direct reading of the default values
* Rearrange the device dispose to avoid confusing messages
* Fix no bridge found issue after changing the bridge configuration

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

kaikreuzer commented:
This PR shows many "full" replacements of classes instead of just small diffs - are the changes that large or did the automatic diff detection not work nicely when creating the PR?

marcelrv commented:
Indeed it is strange that it saw it as full replacements, even the
unchanged copy rights headers are seen as changed.
Don't know why. Only the discovery one had very significant changes as I
moved  the logic.

Any suggestion how I can undo that without major rework.? I squashed
several commits into 1, could that be the reason?
Op 2 jan. 2015 17:36 schreef "Kai Kreuzer" <notifications@github.com>:

> This PR shows many "full" replacements of classes instead of just small
> diffs - are the changes that large or did the automatic diff detection not
> work nicely when creating the PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab2/pull/91#issuecomment-68540069>.
>

kaikreuzer commented:
> Any suggestion how I can undo that without major rework.?

No, I don't have a clue how that can be undone - so let's leave it and live with the big change in history.
Does your comment on #94 mean that you will add another change to this PR? You can do so if you like, I probably won't review it before next week.

marcelrv commented:
Okay, was struggling to make this PR mergable :( in the end I re-applied all the changes once more. Sure there is a smarter way, but that's for another time to find out...

Anyway, seems now to be complete,  #94 included as well and now it also properly sees the changes instead of a mass-change.

kaikreuzer commented:
Thanks Marcel, this looks much nicer now!

I only have one further change request: A fix for the Configuration class has just been merged (see https://github.com/eclipse/smarthome/pull/141), so that you can now retrieve your configuration values in the appropriate type - so you will not need the class casts to String in the initialize() method anymore, which should lead to a much cleaner code (no need to flll your own MaxCubeBridgeConfiguration instance anymore).

marcelrv commented:
@kaikreuzer Yes, sure, no problem.
Will you update the included ESH to the latest, otherwise it won't work for anyone

kaikreuzer commented:
> Will you update the included ESH to the latest, otherwise it won't work for anyone

Also the latest change ("int" support) is now available when you update the target platform. Do you want to change your params back from Integer to int or shall we keep it as it is?

marcelrv commented:
Hi Kai,

Indeed I saw it was included.
It works as it is right now as far as I can see.
Dunno if there is anything against using it as is

regards,
Marcel

2015-01-06 13:24 GMT+01:00 Kai Kreuzer <notifications@github.com>:

> Will you update the included ESH to the latest, otherwise it won't work
> for anyone
>
> Also the latest change ("int" support) is now available when you update
> the target platform. Do you want to change your params back from Integer to
> int or shall we keep it as it is?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab2/pull/91#issuecomment-68858999>.
>

kaikreuzer commented:
Thanks, all is merged now!

kaikreuzer commented on addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxCubeBridgeDiscovery.java:
```diff
@@ -85,6 +77,148 @@ public boolean isBackgroundDiscoveryEnabled() {
+			}
+		};
+		thread.start();
+		receiveDiscoveryMessage();
```
@marcelrv , I just noticed that this line makes the openHAB startup hanging for 8-10 seconds, if no cube is available in the network. Please note that the startBackgroundDiscovery() method should return immediately and not be long-running in any situation. Could you come up with some fix for this?

marcelrv replied:
Hi @kaikreuzer 
Yes, can fix that. I can't just move it in the thread above as that thread is making sure that the receive process (receiveDiscoveryMessage()) is stopped after a while (with the discoveryRunning = false;). Maybe I just start it in a separate thread. Anyway, I'll find something and submit a PR

kaikreuzer replied:
Thanks!

marcelrv replied:
PR #159 is submitted to address this.
